cmd/konvoy-image/cmd/args.go

```diff
@@ -16,10 +16,52 @@ func addClusterArgs(fs *pflag.FlagSet, kubernetesVersion, containerdVersion *str
 }
 
 func addAWSUserArgs(fs *pflag.FlagSet, userArgs *app.UserArgs) {
-	fs.StringVar(&userArgs.AWSBuilderRegion, "region", "", "the aws region to run the builder")
-	fs.StringArrayVar(&userArgs.AMIRegions, "ami-regions", []string{}, "a list of regions to publish amis")
-	fs.StringVar(&userArgs.SourceAMI, "source-ami", "", "a specific ami available in the builder region to source from")
-	fs.StringVar(&userArgs.AMIFilterName, "source-ami-filter-name", "", "a ami name filter on for selecting the source image")
-	fs.StringVar(&userArgs.AMIFilterOwner, "source-ami-filter-owner", "", "only search AMIs belonging to this owner id")
-	fs.StringVar(&userArgs.AWSInstanceType, "aws-instance-type", "", "an instance type available in the builder region to work on")
+	fs.StringVar(
+		&userArgs.AWSBuilderRegion,
+		"region",
+		"",
+		"the aws region to run the builder",
+	)
+	fs.StringArrayVar(
+		&userArgs.AMIRegions,
+		"ami-regions",
+		[]string{},
+		"a list of regions to publish amis",
+	)
+	fs.StringVar(
+		&userArgs.SourceAMI,
+		"source-ami",
+		"",
+		"a specific ami available in the builder region to source from",
+	)
+	fs.StringVar(
+		&userArgs.AMIFilterName,
+		"source-ami-filter-name",
+		"",
+		"a ami name filter on for selecting the source image",
+	)
+	fs.StringVar(
+		&userArgs.AMIFilterOwner,
+		"source-ami-filter-owner",
+		"",
+		"only search AMIs belonging to this owner id",
+	)
+	fs.StringVar(
+		&userArgs.AWSInstanceType,
+		"aws-instance-type",
+		"",
+		"an instance type available in the builder region to work on",
+	)
+	fs.StringArrayVar(
+		&userArgs.AMIUsers,
+		"ami-users",
+		[]string{},
+		"a list AWS user accounts which are allowed use the image",
```

Comment:
IIUC, to use multiple values, you can't pass a comma-separate list, but instead must invoke the flag multiple times. In that case, `--ami-user` makes more sense, e.g.,:
```
--ami-user foo --ami-user bar
```

I think the same reasoning applies to `--ami-groups`. It also applies to `--overrides`, but changing that would break compatibility, I suppose.

Reply:
@dlipovetsky Yeah, I agree. This happened originally due to my limited understanding of how cobra handled StringArray variables. I'll follow this PR and replace them with a StringSlice, which expects a "," separated string for a  value.

Reply:
I'm right there with you. Very unintuitive :disappointed: